core/domain/feedback_services.py

```diff
@@ -368,7 +421,123 @@ def _get_thread_from_model(thread_model):
         thread_model.id, thread_model.exploration_id, thread_model.state_name,
         thread_model.original_author_id, thread_model.status,
         thread_model.subject, thread_model.summary, thread_model.has_suggestion,
-        thread_model.created_on, thread_model.last_updated)
+        thread_model.message_count, thread_model.created_on,
+        thread_model.last_updated)
+
+
+def set_message_count_for_thread(full_thread_id, message_count):
```

Comment:
I think this should be private to the MR job and perhaps even just inlined within it. In normal operation you don't want devs to call this.